### PR TITLE
fix: return correct count of ErrNotExecuted

### PR DESCRIPTION
**Pull request by davidby-influx**
executeQuery() iterates over statements until each is
processed or if an error is encountered that causes
the loop to exit pre-maturely. It should return
ErrNotExecuted for each remaining statement in the
query

closes https://github.com/influxdata/influxdb/issues/19136

- [X] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [X] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [X] Rebased/mergeable
- [X] Tests pass

### CODE REVIEW

File: query/executor.go [hunk 1]
```diff
@@ -380,7 +380,7 @@ LOOP:
 	}
 
 	// Send error results for any statements which were not executed.
-	for ; i < len(query.Statements)-1; i++ {
+	for i++; i < len(query.Statements); i++ {
```
Comment (lesam) [inline query/executor.go:383]: I ran your new tests and they pass without this change, which is concerning

**Review by lesam — CHANGES_REQUESTED** [review]

File: query/executor.go (hunk 1 above)
Comment (davidby-influx) [reply query/executor.go:383]: Have simplified tests conditions, complicated test failure modes. Now can differentiate between previous and new code, as well as correcting a few other small bugs.

File: query/executor_test.go [hunk 2]
```diff
@@ -478,6 +487,66 @@ func TestQueryExecutor_Panic(t *testing.T) {
...
+	testFn := func(testName string, i int, failIndex *int) {
+		*failIndex = i
+		executorCallCount = 0
+		results := e.ExecuteQuery(q, query.ExecutionOptions{}, closing)
+
+		checkNotExecutedResults(t, results, testName, *failIndex, len(q.Statements))
+	}
+	for i := 0; i < len(q.Statements); i++ {
+		closing = make(chan struct{})
+		testFn("executor", i, &executorFailIndex)
```
Comment (lesam) [inline query/executor_test.go:529]: nitpick: I'd prefer to see:

```
executorFailIndex = i
testFn("executor", i)
```

File: query/executor_test.go [hunk 3]
```diff
@@ -478,6 +487,66 @@ func TestQueryExecutor_Panic(t *testing.T) {
...
+}
+
+func checkNotExecutedResults(t *testing.T, results <-chan *query.Result, testName string, failIndex int, lenQuery int) {
+	r := 0
+	cnt := 0
+	for result := range results {
+		if result.Err == query.ErrNotExecuted {
+			cnt++
+			if result.StatementID <= failIndex {
+				t.Fatalf("StatementID for ErrNotExecuted is wrong index: %d", result.StatementID)
```
Comment (lesam) [inline query/executor_test.go:540]: Should we just check that all the indexes we expect to fail are actually failing (build a list of fail indexes and compare to the expected list?) Here we could be missing if one index shows up twice and another is missing.

File: query/executor_test.go [hunk 4]
```diff
@@ -478,6 +487,66 @@ func TestQueryExecutor_Panic(t *testing.T) {
...
+			}
+		}
+		r++
```
Comment (lesam) [inline query/executor_test.go:543]: Looks like `r` is unused?

**Review by lesam — APPROVED** [review]

**Pull request merged**